lib/seen_subject_remover.rb

```diff
@@ -0,0 +1,69 @@
+class SeenSubjectRemover
+
+  attr_reader :user, :workflow, :append_ids, :curr_queue_ids
+
+  def initialize(user, workflow, append_ids)
+    @user = user
+    @workflow = workflow
+    @append_ids = append_ids
+  end
+
+  def ids_to_enqueue
+    append_ids - dup_seen_before_ids
+  end
+
+  private
+
+  def dup_seen_before_ids
+    return @seen_before if @seen_before
+    notify_dup_subject_seen_before_error
+    @seen_before = seen_before_set.pluck(&:id)
+  end
+
+  def append_ids_size
+    append_ids.length
+  end
+
+  def seen_before_set
+    @seen_before ||= if user_seen_subject
+      SetMemberSubject.where(id: append_ids)
+        .joins(:subject)
+        .where(subjects: { id: user_seen_subject.subject_ids })
```

Comment:
i think @marten was talking about plucking here to create the array and then testing in the error_params if seen_before_set.size > 0